GroundDBMemoryPool/src/rdma/util.cc

```diff
@@ -517,12 +415,72 @@ int resources_create(struct resources *res, const char *serverName, const int tc
             ibv_free_device_list(dev_list);
             dev_list = NULL;
         }
-        if (res->sock >= 0)
+    }
+    return rc;
+}
+/******************************************************************************
+ * Function: register_mr
+ *
+ * Input
+ * res pointer to resources structure
+ * buf pointer to the memory segment
+ * size size of the memory segment
+ *
+ * Output
+ * none
+ *
+ * Returns
+ * 0 on success, error code on failure
+ *
+ * Description
+ * Register a new memory region.
+ ******************************************************************************/
+int register_mr(struct resources *res, char* buf, size_t size){
+    int mr_flags = 0;
+    int rc = 0;
+    res->memregs.emplace_back();
+    auto& memreg = res->memregs.back();
+    /* allocate the memory buffer that will hold the data */
+    if (buf == nullptr){
```

Comment:
My understanding on multi-conn is limited, so correct me if I am wrong. 

If `buf != nullptr`, the `buf` is already registered and is held in a different region, so why would `register_mr` be still necessary? Since the buf is registered, we can just add a connection to that region.

Reply:
If `buf != nullptr`, the `buf` is just a regular local memory segment and it has not been registered in the RDMA NIC. Naming here might be confusing. `register_mr()` means creating a new `ibv_mr`, which manages a memory region registered in RDMA NIC.